src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/TimeSpanExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+    internal static class TimeSpanExtensions
+    {
+        /// <summary>
+        /// This method calculates a random exponential backoff which lies between <see cref="RefreshConstants.DefaultCacheExpirationInterval"/> and <see cref="RefreshConstants.DefaultMaxBackoff"/>.
+        /// </summary>
+        /// <param name="interval">The maximum backoff to be used if <paramref name="interval"/> is less than <see cref="RefreshConstants.DefaultMaxBackoff"/>.
+        /// If <paramref name="interval"/> is less than <see cref="RefreshConstants.DefaultCacheExpirationInterval"/>, <paramref name="interval"/> is returned.
+        /// </param>
+        /// <param name="attempts">The number of attempts made to backoff.</param>
+        /// <returns>The calculated exponential backoff time, or <paramref name="interval"/> if it is less than <see cref="RefreshConstants.DefaultCacheExpirationInterval"/>.</returns>
+        public static TimeSpan CalculateBackoffTime(this TimeSpan interval, int attempts)
+        {
+            if (attempts < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(attempts), attempts, "The number of attempts should not be less than 1.");
+            }
+
+            if (interval < RefreshConstants.DefaultCacheExpirationInterval)
```

Comment:
>DefaultCacheExpirationInterval [](http://example.com/codeflow?start=44&length=30)

Do you think if it makes sense to define a separate DefaultMinBackoff for 30s?

Reply:
I like the central place for all these consts. Maybe we can have a more generic naming of the class if you can think of one, or we wait till the "fallback" code change to update it to something else :)

Reply:
Ok, leaving it in RefreshConstants for now. We can update it in future depending on the final implementation of fallback code.